src/modules/tools/laser/Laser.cpp

```diff
@@ -103,11 +106,77 @@ void Laser::on_module_loaded()
 
     //register for events
     this->register_for_event(ON_HALT);
+    this->register_for_event(ON_GCODE_RECEIVED);
+    this->register_for_event(ON_CONSOLE_LINE_RECEIVED);
 
     // no point in updating the power more than the PWM frequency, but no more than 1KHz
     THEKERNEL->slow_ticker->attach(std::min(1000UL, 1000000/period), this, &Laser::set_proportional_power);
 }
 
+void Laser::on_console_line_received( void *argument )
+{
+    if(THEKERNEL->is_halted()) return; // if in halted state ignore any commands
+
+    SerialMessage new_message = *static_cast<SerialMessage *>(argument);
```

Comment:
This way involves a struct copy (SerialMessage has std::string, so its likely copy will allocate more memory) .

Why not keep this a pointer?